Start on a CSS pipeline with LESS

cheshire137 opened the pull request:
This branch adds:
- Travis CI setup for tests and style checker
- LESS for styling
- Fixes style checker complaints
- Organizes code into src/ directory

Hmm, so the app starts, but the styling doesn’t seem to be applied. Any thoughts on this, @probablycorey?

cheshire137 committed:
Add style checker to npm test

cheshire137 committed:
Set a better page title

cheshire137 committed:
Add a .compilerc for Electron to build our LESS

As seen in
https://github.com/electron/electron-compile#how-do-i-set-up-babel--less
--whatever-the-way-i-want.

cheshire137 committed:
Move JS and HTML stuff into src/

cheshire137 committed:
Update package scripts to point at src/main.js

cheshire137 committed:
Add some LESS styling

cheshire137 committed:
Don't need .babelrc when we have .compilerc

probablycorey commented on .compilerc:
```diff
@@ -0,0 +1,9 @@
+{
```
Is this a `electron-compile` thing? It's pretty nice.

cheshire137 replied:
Yep! See https://github.com/electron/electron-compile#how-do-i-set-up-babel--less--whatever-the-way-i-want

probablycorey commented on src/index.html:
```diff
@@ -2,7 +2,8 @@
     <meta charset="UTF-8">
-    <title>Hello World!</title>
+    <title>GitHub Notifications Snoozer</title>
+    <link rel="stylesheet" type="text/less" href="app.less">
```
Maybe this should be `src/app.less` or `./src/app.less`?

cheshire137 replied:
Hmm, just tried both, no luck. Still a white background everywhere!

probablycorey commented on src/main.js:
```diff
@@ -15,7 +15,7 @@ function createWindow () {
   mainWindow.loadURL(`file://${__dirname}/index.html`)
 
-  // mainWindow.webContents.openDevTools()
+  mainWindow.webContents.openDevTools()
```
**protip:** You can press `cmd+opt+i` to open up the console

cheshire137 replied:
Ah yeah, I was doing that then got lazy. I'll comment this back out, didn't intend to check it in!

cheshire137 committed:
Merge branch 'master' into css-for-the-people

cheshire137 committed:
Update readme with env instructions

cheshire137 committed:
Comment out opening dev tools on app start

cheshire137 committed:
Update eslint command to lint scripts/

cheshire137 committed:
Fix some style linter complaints, adjust rules

cheshire137 committed:
Fix bad test runner setup

cheshire137 committed:
Allow running just mocha by itself

cheshire137 commented:
I spoke with @zeke about this yesterday, he thought he might be able to help!

zeke commented:
👋 I haven't figured out the less issue yet, but I did notice that `electron` is in the dependencies, which could be screwing things up. That package name is a different project. Fortunately, that user has graciously handed over the `electron` npm name to us, but we haven't yet published something like `electron-prebuilt` in its place.

cheshire137 committed:
Merge branch 'master' into css-for-the-people

cheshire137 committed:
Restore .babelrc to fix failing test

cheshire137 commented:
Hm, I got this branch updated with master just now and `electron` is no longer a dependency in package.json.

cheshire137 committed:
Add a Travis CI config file

Also don’t do style checker as part of tests yet since it’s failing.

zeke commented:
The Less configuration seems right, but I'm not seeing the styles.

I tried debugging with `DEBUG=*,-babel npm start`, but didn't see any issues in the logs.

@paulcbetts do you see any obvious signs of misconfiguration here?

cheshire137 committed:
Specify a node_js version in Travis CI config

cheshire137 committed:
Move app.less into App component directory

cheshire137 committed:
Don't specify node_modules path in scripts

anaisbetts commented:
I'll have a look in a hot second

cheshire137 committed:
Specify some more Travis CI config stuff

cheshire137 committed:
Fix linter complaints in app-test

cheshire137 committed:
Run both style checks and unit tests with Travis

cheshire137 committed:
Fix eslint path for tests

cheshire137 committed:
Fix linter complaints with verify-token script

cheshire137 committed:
Add eslint-plugin-jsx-a11y

To fix this error on Travis:

```
> eslint -c .eslintrc *.js scripts/* src/**/*.jsx test/**/*.js
Oops! Something went wrong! :(
ESLint couldn't find the plugin "eslint-plugin-jsx-a11y". This can
happen for a couple different reasons:
1. If ESLint is installed globally, then make sure
eslint-plugin-jsx-a11y is also installed globally. A globally-installed
ESLint cannot find a locally-installed plugin.
2. If ESLint is installed locally, then it's likely that the plugin
isn't installed correctly. Try reinstalling by running the following:
    npm i eslint-plugin-jsx-a11y@latest --save-dev
```

cheshire137 committed:
Add eslint-plugin-react

To fix this error on Travis:

> eslint -c .eslintrc *.js scripts/* src/**/*.jsx test/**/*.js
Oops! Something went wrong! :(
ESLint couldn't find the plugin "eslint-plugin-react". This can happen
for a couple different reasons:
1. If ESLint is installed globally, then make sure eslint-plugin-react
is also installed globally. A globally-installed ESLint cannot find a
locally-installed plugin.
2. If ESLint is installed locally, then it's likely that the plugin
isn't installed correctly. Try reinstalling by running the following:
    npm i eslint-plugin-react@latest --save-dev

cheshire137 committed:
Add eslint-plugin-import

To fix this error on Travis:

> eslint -c .eslintrc *.js scripts/* src/**/*.jsx test/**/*.js
Oops! Something went wrong! :(
ESLint couldn't find the plugin "eslint-plugin-import". This can happen
for a couple different reasons:
1. If ESLint is installed globally, then make sure eslint-plugin-import
is also installed globally. A globally-installed ESLint cannot find a
locally-installed plugin.
2. If ESLint is installed locally, then it's likely that the plugin
isn't installed correctly. Try reinstalling by running the following:
    npm i eslint-plugin-import@latest --save-dev

cheshire137 committed:
Add build status to readme

cheshire137 committed:
Use xvfb-maybe for Travis electron-mocha tests

cheshire137 committed:
Disable react/prefer-stateless-function linter

cheshire137 committed:
Cache apt stuff for Travis CI

cheshire137 committed:
Use ./node_modules/.bin/ in scripts

Because suddenly Travis CI cares?

cheshire137 committed:
Add some more eslint customizations

cheshire137 committed:
Specify cached directories on Travis

cheshire137 committed:
Add default npm install to new install section

anaisbetts commented on src/components/App/App.less:
```diff
@@ -0,0 +1,5 @@
+@import "../Filter/Filter.less";
```
It's probably easier to just include this inline in the JSX file:

``` html
<style type="text/less">
@import '../filter/filter.less';
...
</style>
```

anaisbetts commented:
@cheshire137 Ok, tracked this down, it turns out I am not good at computer

anaisbetts commented:
Ok! Blow away `node_modules` and reinstall so that you get `electron-compilers@3.1.6` and this should just work. You should also go into $TMPDIR and remove any folder that starts with `compileCache`

cheshire137 commented:
Thank you, it works!

![screen shot 2016-07-20 at 1 22 50 pm](https://cloud.githubusercontent.com/assets/82317/16999888/72d94164-4e7d-11e6-84e9-5fd2269b22e0.png)

I checked out [your commits](https://github.com/electron/electron-compilers/compare/8b77d4cb5ed5c3843f6efcc3fb158f442e053f92...3d9498f2dbd7da1e7c64605d0235ce3101f572ae)--thanks for helping!

cheshire137 committed:
Don't actually do magenta and green backgrounds

zeke commented:
👏

probablycorey commented on package.json:
```diff
@@ -12,13 +12,19 @@
     "eslint-config-airbnb": "^9.0.1",
+    "eslint-plugin-import": "^1.11.0",
+    "eslint-plugin-jsx-a11y": "^2.0.1",
+    "eslint-plugin-react": "^5.2.2",
```
Do we need these three packages? I think they are included with the `eslint-config-airbnb` package's dependencies.

cheshire137 replied:
We do! Check the full commit messages, tests failed on Travis till these were added.

anaisbetts replied:
Is Travis running old-ass npm 2 or something?

cheshire137 replied:
Haha I think my Travis config file specifies node 6. Check a build, it spits out node, npm, and nvm version numbers.
